### ISSUE TITLE
Update portfinder to handle bash on windows error

### ISSUE THREAD

**Issue by eriktrom** [ember-cli/ember-cli#6364]
Upstream merged PR: https://github.com/indexzero/node-portfinder/pull/46

Fixes one of (one or many, TBD) issues filed at #6338 

Also, I found these comments @ the underlying bash on windows issue that mentions the `os.networkInterfaces()` error, specifically for ember-cli and angular-cli users:

https://github.com/Microsoft/BashOnWindows/issues/468#issuecomment-252081810
https://github.com/Microsoft/BashOnWindows/issues/468#issuecomment-247522805
https://github.com/Microsoft/BashOnWindows/issues/1093

One question here - earlier today @nw [mentioned](https://github.com/indexzero/node-portfinder/issues/45) that we should apply semantic versioning to node-portfinder and put any windows bugs into 2.0.0. I agree, especially for the windows 10 native hack, which broke angular-cli and ember-cli yesterday. 

However, in this simpler [and seemingly widely used hack](https://github.com/yarnpkg/yarn/pull/772/files) which is very similar to upstream portfinder hack(pr 46, linked above) should we ship that in portfinder 2.0.0 or 1.10.0?

Let me know thoughts and consensus here. Thanks!

**Comment by nw** [comment]
I think a minor release is appropriate (1.10.0).

The PR looks good, I'm just not a fan of 'throw' in utility modules. However, you could argue using this lib and not returning a valid value should crash. I generally like to leave that decision to the app dev. I'm 50/50 on throw here.

**Comment by eriktrom** [comment]
@nw - regarding the return value. I agree, the reason is b/c it's explicit and self documenting. Actually had it this way in 1.0.8(not buggy loc), I'll just return what's in the docs and remove the comment pointing to a the default.

One q: Do u agree that '0.0.0.0' and '::' per the node docs default value or something better. I ask b/c the only reason for the for loops over 'os.networkInterfaces' became a last resort hack which is not specific and 'handle the error as it comes' oriented. 

I really would like to make the next version of portfinder correct. Any feedback is very much appreciated. 

Hey thanks. I really appreciate the 'add some robustness' feedback.

**Comment by eriktrom** [comment]
Sorry and regarding the error - don't u think I should re-throw unless I handle the error explicitly? Feel feee to dive into that as well. Totally curious. Thanks again.

**Comment by eriktrom** [comment]
So an update here - I had some odd results while testing the bash support this morning. I am also not in town(and don't want to ship this rushed). 

I'll be back on the 1st and I'll re-open this PR - at which time I'll work through why I was getting portfinder to correctly run the tests an hour ago and now can't reproduce that now. I have 3 windows users now and the shells seem to be acting up - when I run the tests now I get nothing as output.

Related to this, when things we're working an hour or so ago, aka, when npm linkning portfinder -> ember-cli -> global node modules, then running ember new, I got the following errors (relates to #46)
https://gist.github.com/532ad837acfd6f03b4b4e819ed9b0fb2

tl;dr - when I wrote that gist, portfinder was working - I could tell this b/c before I npm linked the ember-cli with it, running `ember new blah` actually threw the `os.networkInterfaces` error - as did `ember s -p 0`. Regardless, we still have other issues to hash out before this will work anyway. 

Between that and the tests that I need to fix in order for the bash shell to skip the socket tests (that use file descriptors, also not implemented in windows on bash) I am going to opt to wait until I'm back home and have ample time to sit down with this. (This will also prevent me from making terrible comments here on my phone, as in the case of last nite :) )

**Comment by eriktrom** [comment]
based on feedback, starting with this comment from portfinder - https://github.com/indexzero/node-portfinder/issues/39#issuecomment-258681044, I will make a release of portfinder in the morning that provides a fix to make it work with bash on windows

ember-cli will not fully work in this env due to issues in my gist found above (first comment) (relating to https://github.com/ember-cli/ember-cli/issues/6338), but other projects are soon to need it (like webpack)

big shout out to @khrome83 for helping me verify portfinder's next release works on a native windows machine :)

**Comment by nathanhammond** [comment]
@eriktrom It sounds like this release will be backward compatible but doesn't 100% address all issues that can appear on Windows + BASH subsystem, confirm?

**Comment by eriktrom** [comment]
@nathanhammond - confirm - the rest of the errors we'll have to deal with case by case - e.g., `Error: EISDIR: illegal operation on a directory, read` <-- what is that? LOL 

will cut a release of portfinder soon here

**Comment by eriktrom** [comment]
portfinder 1.10.0 has been released (before a holiday, in case rollback is needed, no one gets hurt :)
